### ISSUE TITLE
Can't find interface-file declaration for variable id

### ISSUE THREAD

**Issue by falsandtru** [haskell/haskell-ide-engine#1100]
HIE stops many times.

```
hie-8.6.3: hie-8.6.3: panic! (the 'impossible' happened)
  (GHC version 8.6.3 for x86_64-unknown-linux):
	lookupGlobal

Can't find interface-file declaration for variable id
  Probable cause: bug in .hi-boot file, or inconsistent .hi file
  Use -ddump-if-trace to get an idea of which file caused the error
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler/utils/Outputable.hs:1160:37 in ghc:Outputable
      pprPanic, called at compiler/typecheck/TcEnv.hs:132:32 in ghc:TcEnv
```

**Comment by fendor** [comment]
This looks like a ghc bug?
Did you try it on other projects as well?
Also, what OS are you on and can you build the project via stack or cabal? Did you try cleaning the caches?

**Comment by falsandtru** [comment]
Not sure but looks like a bug of HIE. I've seem some errors (the other errors are not reported yet) *in writing code to import modules* with https://github.com/falsandtru/haskell-examples on Ubuntu 18.04.

**Comment by fendor** [comment]
On the linked project, HIE works. 
Did you try cleaning all `.stack-work` folders, as well as `dist` and `new-dist`?
Also, what is your `stack --version` output?
It doesnt really look like an error by HIE.

**Comment by falsandtru** [comment]
Close the issue for a time.

**Comment by dcastro** [comment]
@falsandtru, are you still seeing this error, or did you find a solution?
I get this error and hie crashes every ~10 minutes or so.
Tried removing `.stack-work`, didn't help.

MacOS 10.14 / Stack 1.9.3 / lts-13.7 / GHC 8.6.3 / Using vscode plugin

**Comment by falsandtru** [comment]
This error may be fixed after I resolved #1102 but anyway HIE throws a similar error when I'm touching the unresolved dependent packages.

**Comment by dcastro** [comment]
@falsandtru thanks for the reply. I've updated to the latest version of hie. it seems to still be happening, but a lot less frequently it seems (just once in the past day)

[
![Screenshot 2019-04-04 at 13 07 53](https://user-images.githubusercontent.com/718166/55554515-dfda5d80-56da-11e9-91f0-b8d12a782b32.png)
](url)

**Comment by falsandtru** [comment]
Yes. HIE still has a lot of bugs. We have to restart our editor to clear incorrect errors.

**Comment by fendor** [comment]
Whenever you upgrade a stack ghc version, I recommend a `stack clean`. 
Does the project build via `stack build`?

**Comment by enobayram** [comment]
@fendor Thanks for your comment, I had run `stack clean` and `rm -rf .stack-work` many times since the upgrade, but I've just tried `rm -rf $(find . -name .stack-work)` just in case. I didn't get get the error since then, I'll report back if I do.

**Comment by enobayram** [comment]
Just happened again:
```
Can't find interface-file declaration for type constructor or class Day
  Probable cause: bug in .hi-boot file, or inconsistent .hi file
  Use -ddump-if-trace to get an idea of which file caused the error
Call stack:
    CallStack (from HasCallStack):
      callStackDoc, called at compiler/utils/Outputable.hs:1160:37 in ghc:Outputable
      pprPanic, called at compiler/typecheck/TcEnv.hs:132:32 in ghc:TcEnv

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

Maybe it's relevant; this happened shortly after adding `hashable-time` to the dependencies in the project `.cabal`, note how the error message involves `Day`.
